text/0000-native-graphql-source.md

```diff
@@ -0,0 +1,156 @@
+    {
+      resolve: "gatsby-source-graphql",
+      options: {
+        // This type will contain remote schema Query type
+        typeName: "SWAPI",
+        // This is field under which it's accessible
+        fieldName: "swapi",
+        // Url to query from
+        url: "https://api.graphcms.com/simple/v1/swapi",
+      },
+    },
+    // Passing paramaters (passed to apollo-link)
+    {
+      resolve: "gatsby-source-graphql",
+      options: {
+        typeName: "GitHub",
+        fieldName: "github",
+        // Url to query from
+        url: "https://api.github.com/graphql",
+        // HTTP headers
+        headers: {
+          Authorization: "bearer <GITHUB_TOKEN>",
+        },
+        // Additional options to pass to node-fetch
+        fetchOptions: {},
+      },
+    },
+    // Creating arbitrary Apollo Link (for advanced situations)
+    {
+      resolve: "gatsby-source-graphql",
+      options: {
+        typeName: "GitHub",
+        fieldName: "github",
+        // Create Apollo Link manually. Can return a Promise.
+        createLink: (pluginOptions) => {
+          return createHttpLink({
+            uri: 'https://api.github.com/graphql',
+            headers: {
+              'Authorization': `bearer ${process.env.GITHUB_TOKEN}`,
+            },
+            fetch,
+          })
+      },
+    },
+  ],
+}
+```
+
+Querying it
+
+```graphql
+{
+  # Field name parameter defines how you can access third party api
+  swapi {
+    allSpecies {
+      name
+    }
+  }
+  github {
+    viewer {
+      email
+    }
+  }
+}
+```
+
+Low-level Gatsby API for future plugins
+
+```js
+exports.sourceNodes = async ({ boundActionCreators }) => {
+  const { addThirdPartySchema } = boundActionCreators
+  addThirdPartySchema(yourGraphQLJSExecutableSchema)
+})
+```
+
+# Motivation
+
+When Gatsby started, there weren't that many GraphQL APIs in the wild. Nowadays, not only there are many public GraphQL APIs, but also many people have their own internal GraphQL API. Tools like Prisma and AWS Appsync are increasingly popular. As Gastby uses GraphQL, it's weird that there is no native way to use those plugin schemas. Currently one needs to create a custom source plugin for every GraphQL API.
+
+This RFCs proposes the way to automatically merge Gatsby schema with third-party APIs. In addition, it provides a simple to use plugin to do that, that should cover most of the use cases.
+
+Low-level API should also enable a more deep customization of Gatsby Schemas, by exposing almost full power of graphql-tools schema-stitching. This probably will allow more advanced tools that replace or modify some built-in types.
+
+# Detailed design
+
+## Action Creator
+
+`addThirdPartySchema(schema: GraphQLSchema)`
+
+Adds a schema to third-party schema list. At schema creation time all the schemas are merged with [schema stitching](https://www.apollographql.com/docs/graphql-tools/schema-stitching.html)
+
+## graphql-source-graphql
```

Comment:
Shouldn't this be `gatsby-source-graphql` as in the examples above, or is this another plugin?